plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java

```diff
@@ -0,0 +1,128 @@
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static org.junit.jupiter.api.Assumptions.abort;
+
+@Execution(ExecutionMode.SAME_THREAD) // Custom hive4 image doesn't support running concurrent queries
+class TestHive4OnDataLake
+        extends BaseTestHiveOnDataLake
+{
+    private String bucketName;
+    private Hive4MinioDataLake hiveMinioDataLake;
+    private HiveMetastore metastoreClient;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.bucketName = "test-hive-insert-overwrite-" + randomNameSuffix();
+        this.hiveMinioDataLake = closeAfterClass(new Hive4MinioDataLake(bucketName));
+
+        this.hiveMinioDataLake.start();
+        this.metastoreClient = new BridgingHiveMetastore(
+                testingThriftHiveMetastoreBuilder()
+                        .metastoreClient(this.hiveMinioDataLake.getHiveMetastore().getHiveMetastoreEndpoint())
+                        .build(this::closeAfterClass));
+        return S3HiveQueryRunner.builder(hiveMinioDataLake)
+                .addExtraProperty("sql.path", "hive.functions")
+                .addExtraProperty("sql.default-function-catalog", "hive")
+                .addExtraProperty("sql.default-function-schema", "functions")
+                .setHiveProperties(
+                        ImmutableMap.<String, String>builder()
+                                .put("hive.insert-existing-partitions-behavior", "OVERWRITE")
+                                .put("hive.non-managed-table-writes-enabled", "true")
+                                // Below are required to enable caching on metastore
+                                .put("hive.metastore-cache-ttl", "1d")
+                                .put("hive.metastore-refresh-interval", "1d")
+                                // This is required to reduce memory pressure to test writing large files
+                                .put("s3.streaming.part-size", HIVE_S3_STREAMING_PART_SIZE.toString())
+                                // This is required to enable AWS Athena partition projection
+                                .put("hive.partition-projection-enabled", "true")
+                                .buildOrThrow())
+                .build();
+    }
+
+    @Override
+    Minio minio()
+    {
+        return hiveMinioDataLake.getMinio();
+    }
+
+    @Override
+    String bucketName()
+    {
+        return bucketName;
+    }
+
+    @Override
+    HiveMetastore metastoreClient()
+    {
+        return metastoreClient;
+    }
+
+    @Override
+    MinioClient minioClient()
+    {
+        return hiveMinioDataLake.getMinioClient();
+    }
+
+    @Override
+    String runOnHive(String query)
+    {
+        return hiveMinioDataLake.getHiveServer().runOnHive(query);
+    }
+
+    @Override
+    @Test
+    public void testSyncPartitionOnBucketRoot()
+    {
+        abort("Fails with `location must not be root path`");
```

Comment:
maybe these should be linked to issues in github (for tracking purposes)

Reply:
Good thought. Linked issues instead now.